### PR TITLE
DOCS-934: Clean up container procedures

### LINKED ISSUE #934: Update or remove docker install tutorials which link to Linux internals
Some of our docker install, upgrade, or mod tutorials are not correctly conditionalized. 

We either need to

a) update those tutorials to be docker specific, and/or

b) Remove them entirely (e.g. multi-node multi-drive or pool expansion)

### PR DESCRIPTION
**Pull request by ravindk89**
Closes #934 

Covers the upgrade page + network TLS page, which was pretty bad.

Does not cover the OIDC/AD/LDAP pages or the troubleshooting pages. Those would require a lot more work than I can allocate right now, and no one has asked for them that I can recall.

Staged: 

- http://192.241.195.202:9000/staging/DOCS-934/container/operations/install-deploy-manage/upgrade-minio-deployment.html
- http://192.241.195.202:9000/staging/DOCS-934/container/operations/network-encryption.html

### CODE REVIEW

File: source/operations/install-deploy-manage/deploy-minio-multi-node-multi-drive.rst [hunk 1]
```diff
@@ -61,24 +61,37 @@ The :ref:`integrations-nginx-proxy` reference provides a baseline configuration
...
 
-You can specify the entire range of hostnames using the expansion notation
-``minio{1...4}.example.com``.
+      198.0.2.10    minio-01.example.net
+      198.51.100.3  minio-02.example.net
+      198.0.2.43    minio-03.example.net
+      198.51.100.12 minio-04.example.net
 
-Configuring DNS to support MinIO is out of scope for this procedure.
+   The above hosts configuration would support expansion notation of ``minio-0{1...4}.example.net``.
```
Comment (feorlen) [inline source/operations/install-deploy-manage/deploy-minio-multi-node-multi-drive.rst:94]: maybe
```suggestion
   The above hosts configuration supports expansion notation of ``minio-0{1...4}.example.net``, mapping the sequential hostnames to the desired IP addresses.
```

File: source/operations/network-encryption.rst [hunk 2]
```diff
@@ -79,6 +79,33 @@ Enabling TLS
...
+   You can use the MinIO :minio-git:`certgen <certgen>` to mint self-signed certificates for enabling TLS for evaluating MinIO with TLS enabled.
+   For example, the following command generates a self-signed certificate with a set of IP and DNS SANs associated to the MinIO Server hosts:
+
+   .. code-block:: shell
+
+      certgen -host "localhost,minio-*.example.net"
+
+   You may need to start the container and set a ``--hostname`` that matches the TLS certificate DNS SAN.
+
+   Move the certificates to the path on your local host machine which mounts to the ``--certs-dir`` directory on the container.
```
Comment (feorlen) [inline source/operations/network-encryption.rst:105]: Maybe this shorter alternative is still accurate?
```suggestion
   Move the certificates to the local host machine path that the container mounts to its ``--certs-dir`` path.
```

File: source/operations/network-encryption.rst [hunk 3]
```diff
@@ -79,6 +79,33 @@ Enabling TLS
...
+   When the MinIO container starts, the server searches the specified location for certificates and  uses them enable TLS.
```
Comment (feorlen) [inline source/operations/network-encryption.rst:106]: duplicate space
```suggestion
   When the MinIO container starts, the server searches the specified location for certificates and uses them enable TLS.
```

File: source/includes/container/steps-upgrade-minio-deployment.rst [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+You can run the ``podman container inspect`` or ``docker inspect`` command to inspect the container and validate the current container image:
+
+.. code-block:: shell 
+   :class: copyable
+
+   # For docker, use docker inspect
+   podman container inspect --format='{{.Config.Image}}' CONTAINER_NAME
+
+If the output resembles the following, you created the container using the latest stable image tag:
```
Comment (feorlen) [inline source/includes/container/steps-upgrade-minio-deployment.rst:39]: Maybe
```suggestion
The following output indicates the container was created using the most recent stable image tag:
```

File: source/includes/linux/steps-upgrade-minio-deployment.rst [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+Upgrades Are Non-Disruptive
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MinIO's upgrade-then-restart procedure does *not* require taking downtime or scheduling a maintenance period.
+MinIO restarts are fast, such that restarting all server processes in parallel typically completes in a few seconds. 
+MinIO operations are atomic and strictly consistent, such that applications using MinIO or S3 SDKs can rely on the built-in :aws-docs:`transparent retry <general/latest/gr/api-retries.html>` without further client-side logic.
+This ensures upgrades are non-disruptive to ongoing operations.
+
+"Rolling" or serial "one-at-a-time" upgrade methods do not provide any advantage over the recommended "parallel" procedure, and can introduce unnecessary complexity to the upgrade procedure.
+For virtualized environments which *require* rolling updates, you should amend the recommend procedure as follows:
```
Comment (feorlen) [inline source/includes/linux/steps-upgrade-minio-deployment.rst:23]:
```suggestion
For virtualized environments which *require* rolling updates, you should modify the recommend procedure as follows:
```

File: source/includes/linux/steps-upgrade-minio-deployment.rst [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+1. Update the MinIO Binary in the virtual machine or container one at a time.
+2. Restart the MinIO deployment using :mc-cmd:`mc admin service restart`.
+3. Update the virtual machine/container configuration to use the matching newer MinIO image.
+4. Perform the rolling restart of each machine/container with the updated image.
+
+Check Release Notes
+~~~~~~~~~~~~~~~~~~~
+
+MinIO publishes :minio-git:`Release Notes <minio/releases>` for your reference as part of identifying the changes applied in each release.
+Review the associated release notes between your current MinIO version and the newer release such that you have a complete view of any changes.
```
Comment (feorlen) [inline source/includes/linux/steps-upgrade-minio-deployment.rst:34]:
```suggestion
Review the associated release notes between your current MinIO version and the newer release so you have a complete view of any changes.
```

**Comment by feorlen** [comment]
Pretend this comment is an approving review. GitHub is having some trouble this morning.

Noted a couple ideas for your consideration.

File: source/includes/linux/steps-upgrade-minio-deployment.rst (hunk 5 above)
Comment (djwfyi) [reply source/includes/linux/steps-upgrade-minio-deployment.rst:23]:
```suggestion
For virtualized environments which *require* rolling updates, you should modify the recommended procedure as follows:
```

File: source/operations/network-encryption.rst (hunk 3 above)
Comment (djwfyi) [reply source/operations/network-encryption.rst:106]:
```suggestion
   When the MinIO container starts, the server searches the specified location for certificates and uses them to enable TLS.
```

**Review by djwfyi — APPROVED** [review]
Couple of minor grammar things, otherwise LGTM.

**Comment by ravindk89** [comment]
OK that should do it. Thanks folks.

**Pull request merged**